app/Http/Controllers/Users/UserFilesController.php

```diff
@@ -34,17 +35,10 @@ public function store(UploadFileRequest $request, User $user)
             $file_name = $request->handleFile('private_uploads/users/', 'user-'.$user->id, $file);
 
             //Log the uploaded file to the log
-            $logAction = new Actionlog();
-            $logAction->item_id = $user->id;
-            $logAction->item_type = User::class;
-            $logAction->created_by = auth()->id();
-            $logAction->note = $request->input('notes');
-            $logAction->target_id = null;
-            $logAction->created_at = date("Y-m-d H:i:s");
-            $logAction->filename = $file_name;
-            $logAction->action_type = 'uploaded';
-
-            if (! $logAction->save()) {
+            $user->setLogFilename($file_name);
+            $user->setLogNote($request->input('notes'));
+
+            if (!$user->logAndSaveIfNeeded(ActionType::Uploaded)) {
                 return JsonResponse::create(['error' => 'Failed validation: '.print_r($logAction->getErrors(), true)], 500);
```

Comment:
We should use the UploadedFilesTransformer for this, or at least our standard API JSON response formatter.